.github/workflows/validate-samples.yml

```
@@ -0,0 +1,76 @@
+
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+    branches: [main, master]
+
+# Least-privilege GITHUB_TOKEN (public repo; CI only needs read + npm cache)
+permissions:
+  contents: read
+  actions: read
+  actions: write
```

actionlint (1.7.12)

[error] key "actions" is duplicated in "permissions" section. previously defined at line:12,col:3. note that this key is case insensitive

(syntax-check)